### ISSUE TITLE
Coding tasks

### ISSUE THREAD

**Issue by NtaylorOX** [NtaylorOX/Public_Prompt_Mimic_III#1]
# For all tasks

## Coding to do
- [x] Add a zero shot function to the prompt_experiment_runner. Which can be optionally run before any training etc.
- [x] Save all of the paramters to a json or yaml file. Will be easier to reload the trained models
- [ ] Add functionality to prompt experiment runner to pass through a standard classification layer without prompts
- [ ]  See if compatible with pytorch-lightning  
- [x] Refactor code to easily allow comparison of multiple baseline neural networks on mimic-iii tasks - in mimic-icd9-classification make a dataset folder for all the mimic data
- [x] Make clinical longformer pipeline save logs/ckpts/params in sensible locations based on PLM freezing
- [x] Implement a mimic few shot sampler for few shot training etc
- [x] Add all the same metrics across all pipelines e.g. macro and weighted f1/precision/recall etc
- [ ]  Look at the lit language interpretability tool and run examples https://github.com/PAIR-code/lit
- [ ] Look at attention/gradient based interpretability approaches e.g. https://colab.research.google.com/github/ml6team/quick-tips/blob/main/nlp/2021_04_22_shap_for_huggingface_transformers/explainable_transformers_using_shap.ipynb for the triage task
- [x] Develop a script to digest and parse tensorboard logs and turn into a queryable csv file/dataframe
- [ ] Try adapt the FARM framework for our tasks - either same as https://github.com/bvanaken/clinical-outcome-prediction or from the notebook provided by FARM : https://github.com/deepset-ai/FARM/blob/master/tutorials/1_farm_building_blocks.ipynb 
- [x] Apply few shot learning processor to the pytorch lightning models?
- [ ]  Adapt the pytorch lightning pipeline to work dynamically with all tasks and label columns 

## Datasets/Tasks

mimic-iii readmission
icd9-diagnosis-cllassification
icd9-triage

and following: 

## Other mimic tasks

Create following benchmark datasets and run all of above:

https://github.com/YerevaNN/mimic3-benchmarks/tree/v1.0.0-alpha

and also this:

https://github.com/bvanaken/clinical-outcome-prediction

### Full training and zero shot together
- [ ] run all tasks through prompt learning pipeline with : mixed_template and manual verbalizer - both frozen plm and finetuned
- [ ] same as above but with pytorch lightning pipeline
### Few-shot with the fewshot sampler
- [ ] run all tasks through prompt learning pipeline with : mixed_template and manual verbalizer - both frozen plm and finetuned
- [ ] same as above but with pytorch lightning pipeline
